src/robotFactory.js

```diff
@@ -1,10 +1,89 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, {
+    x = 0,
+    y = 0,
+  }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x,
+      y,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}`
+    + `, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, {
+    x = 0,
+    y = 0,
+    z = 0,
+  }, chipVersion) {
```

Comment:
```suggestion
  constructor(
    name, 
    weight, 
    {
      z = 0,
      ...baseCoords,
    }, 
    chipVersion,
  ) {
```
Better place each arg on a new line if there are more than 2 ones.
Also, you don't need x and y coords here - use rest operator.

Reply:
How i can use ...baseCoords at here?

![image](https://user-images.githubusercontent.com/116715761/223385114-5b56d390-3bc6-4d6d-9cc8-5a063322613b.png)